native/kotlin/api/kotlin/src/main/kotlin/rs/wordpress/api/kotlin/WpRequestExecutor.kt

```diff
@@ -29,6 +35,44 @@ class WpRequestExecutor(
                 }
             }
 
+            okHttpClient.newCall(requestBuilder.build()).execute().use { response ->
+                return@withContext WpNetworkResponse(
+                    body = response.body?.bytes() ?: ByteArray(0),
+                    statusCode = response.code.toUShort(),
+                    headerMap = WpNetworkHeaderMap.fromMultiMap(response.headers.toMultimap())
+                )
+            }
+        }
+
+    override suspend fun uploadMedia(mediaUploadRequest: MediaUploadRequest): WpNetworkResponse =
+        withContext(dispatcher) {
+            val requestBuilder = Request.Builder().url(mediaUploadRequest.url())
+            val multipartBodyBuilder = MultipartBody.Builder()
+                .setType(MultipartBody.FORM)
+            mediaUploadRequest.mediaParams().forEach { (k, v) ->
+                multipartBodyBuilder.addFormDataPart(k, v)
+            }
+            val file =
+                WpRequestExecutor::class.java.classLoader?.getResource(mediaUploadRequest.filePath())?.file?.let {
```

Comment:
I am not sure if this is the correct way to get the file in Android. I _think_ it's correct, but we might need to pass a function in the initializer so it can be configured.

I am leaving it as is for now and we can address it if we find that it's not working when we integrate it to WordPress-Android.